src/registers.rs

```diff
@@ -0,0 +1,89 @@
+use crate::errors::VMError;
+
+const PC_START: u16 = 0x3000;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum RegisterFlags {
+    Pos = 1 << 0,
+    Zro = 1 << 1,
+    Neg = 1 << 2,
+}
+
+const NUM_REGISTERS: usize = 8; // R0-R7
+
+pub struct Registers {
+    regs: [u16; NUM_REGISTERS],
+    pub pc: u16,
+    condition: RegisterFlags,
+}
+
+impl Registers {
+    pub fn new() -> Self {
+        Self {
+            regs: [0; NUM_REGISTERS],
+            pc: PC_START,
+            condition: RegisterFlags::Zro,
+        }
+    }
+
+    pub fn get(&self, register: usize) -> Result<u16, VMError> {
+        self.regs
+            .get(register)
+            .copied()
+            .ok_or(VMError::InvalidRegister)
+    }
+
+    pub fn set(&mut self, register: usize, value: u16) {
+        if let Some(reg) = self.regs.get_mut(register) {
+            *reg = value;
+        }
+    }
```

Comment:
I think it would be better to panic or at least log the error if you are trying to set to a register that does not exist. It will be useful later when debugging errors.

Reply:
Thx, i will add logging functionality in the next pr when i start debugging